app/presenters/tree_node_builder.rb

```diff
@@ -459,4 +463,63 @@ def miq_report_node(text, image, name, status)
       generic_node(text, image)
     end
   end
+
+  def assigned_server_role_node(object)
+    @node = {
+      :key      => build_object_id,
+      :addClass => "dynatree-title",
+      :title    => options[:tree] == :servers_by_role_tree ?
+        "#{Dictionary.gettext('MiqServer', :type => :model, :notfound => :titleize)}: #{object.name} [#{object.id}]" :
+        "Role: #{object.server_role.description}"
+    }
+
+    if object.master_supported?
+      priority = case object.priority
+                 when 1
+                   "primary, "
+                 when 2
+                   "secondary, "
+                 else
+                   ""
+                 end
+    end
+    if object.active? && object.miq_server.started?
+      @node[:icon] = ActionController::Base.helpers.image_path("100/on.png")
+      @node[:title] += _(" (%{priority}active, PID=%{number})") % {:priority => priority, :number => object.miq_server.pid}
```

Comment:
@mzazrivec how about somethign like this:
``
if object.master_supported?
      priority = case object.priority
                 when 1
                   _("primary, ")
                 when 2
                   _("secondary, ")
                 else
                   ""
                 end
    end
``
because in the next few lines we set state of server role in the title as active/available/unavailable based upon other attributes of server, trying to avoid too many if/elsif